### ISSUE TITLE
Adding union, iter_counts, etc.

### ISSUE THREAD

**Issue by HarrisonMc555** [jmitchell/multiset#17]
I used this crate for a project, but I felt the need to add some more functionality.

Specifically, I added:
  1. The ability to have an iterator over the keys along with the values
  2. A `union` function that combined two `HashMultiSet`s.

If I folded these additions (and possibly similar additions, like `intersection`) into the crate itself would you be open to a pull request?

**Comment by mashedcode** [comment]
I really like the idea of things that are also available in `HashSet` like `union` for example to be implemented. I'm sure contributions are welcome.

**Comment by jmitchell** [comment]
> 1. The ability to have an iterator over the keys along with the values

Good idea. Maybe we can draw inspiration from `HashMap`'s `keys()` iterator method.

For consistency sake, in a future major version bump we could rename the existing `iter()` logic to `values()` and support a `HashMap`-style `iter()` and `iter_mut()` over key-value pairs.

> 2. A union function that combined two HashMultiSets.

Did you happen to notice the `Add` trait implementation? It enables the `+` operator to construct the union of two `HashMultiSet`s. This is similar to `HashSet::union`, except `union` takes references returns a lazy iterator `hash_set::Union` whereas `HashMultiSet`'s `add` consumes and produces _owned_ values. (In hindsight starting with `union` instead of `+` probably would have been better.)

Proposals to add an `std`-inspired `union` are welcome. Alternatives may also be considered if there's clear benefits to deviating from standard conventions.

> would you be open to a pull request?

Yes, thank you.
